// @shopify/cli/dist/node-package-manager-TLCUDNS2.js
import {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
} from "./chunk-FGSRYMBN.js";
import "./chunk-BYPQXSAL.js";
import "./chunk-4DZKSJ2R.js";
import "./chunk-2HGYYNE5.js";
import "./chunk-XVNW332R.js";
import "./chunk-K6Y4FYT5.js";
import "./chunk-OJOHMVV7.js";
import "./chunk-CBBS4CV7.js";
import "./chunk-SNOECVP4.js";
import "./chunk-UBB7JKND.js";
import "./chunk-WNDN5FAY.js";
import "./chunk-7Q3MMWAC.js";
import "./chunk-KYB6A4PE.js";
import "./chunk-NB4NLOEJ.js";
import "./chunk-522OB3EU.js";
import "./chunk-POZ5MGPT.js";
export {
  FindUpAndReadPackageJsonNotFoundError,
  PackageJsonNotFoundError,
  UnknownPackageManagerError,
  addNPMDependencies,
  addNPMDependenciesIfNeeded,
  addNPMDependenciesWithoutVersionIfNeeded,
  addResolutionOrOverride,
  bunLockfile,
  checkForCachedNewVersion,
  checkForNewVersion,
  findUpAndReadPackageJson,
  getDependencies,
  getPackageManager,
  getPackageName,
  getPackageVersion,
  inferPackageManager,
  installNPMDependenciesRecursively,
  installNodeModules,
  lockfiles,
  npmLockfile,
  packageManager,
  packageManagerFromUserAgent,
  pnpmLockfile,
  pnpmWorkspaceFile,
  readAndParsePackageJson,
  usesWorkspaces,
  versionSatisfies,
  writePackageJSON,
  yarnLockfile
};
